app/assets/javascripts/openproject-helper-angular.js

```diff
@@ -0,0 +1,37 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2017 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2017 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+OpenProject.Helpers.Angular = OpenProject.Helpers.Angular || {};
+
+OpenProject.Helpers.Angular.compile = function(htmlCode) {
+  var injector = angular.element(document.body).injector();
+  var compile = injector.get('$compile');
+  var newScope = injector.get('$rootScope').$new();
```

Comment:
Just a warning: This scope will never be destroyed. This is probably okay while we're using this on a page that is likely to be destroyed soon after due to a hard refresh though. An alternative would be to remember this special 'compile-only' scope within the rootScope, and re-use it if used again.